examples/general.c

```diff
@@ -708,10 +709,32 @@ static void config_files(const char *repo_path, git_repository* repo)
 	sprintf(config_path, "%s/config", repo_path);
 	check_error(git_config_open_ondisk(&cfg, config_path), "opening config");
 
-	git_config_get_int32(&j, cfg, "help.autocorrect");
-	printf("Autocorrect: %d\n", j);
+	error_code = git_config_get_int32(&j, cfg, "help.autocorrect");
+	switch (error_code)
+	{
+		case 0:
+			printf("Autocorrect: %d\n", j);
+			break;
+		case GIT_ENOTFOUND:
+			printf("Autocorrect: Undefined\n");
+			break;
+		default:
+			check_error(error_code, "get_int32 failed");
+	}
+	git_config_free(cfg);
 
 	check_error(git_repository_config_snapshot(&snap_cfg, repo), "config snapshot");
-	git_config_get_string(&email, snap_cfg, "user.email");
-	printf("Email: %s\n", email);
+	error_code = git_config_get_string(&email, snap_cfg, "user.email");
+	switch (error_code)
```

Comment:
I was curious how your proposed macro would look so I coded it.  I enclose below the macro and its invocations.  Is this better than the original?  

```
#define GIT_CFG_ERR_CHK(display_name, format, variable, error_msg, error_code) \
    switch (error_code) \
    { \
        case 0: \
            printf(display_name ": " format "\n", variable); \
            break; \
        case GIT_ENOTFOUND: \
            printf(display_name ": Undefined\n"); \
            break; \
        default: \
            check_error(error_code, error_msg); \
	}

    error_code = git_config_get_int32(&j, cfg, "help.autocorrect");
    GIT_CFG_ERR_CHK("Autocorrect", "%d", j, "get_int32 failed", error_code)

    error_code = git_config_get_string(&email, snap_cfg, "user.email");
    GIT_CFG_ERR_CHK("Email", "%s", email, "get_string failed", error_code)
```